src/main/java/org/folio/service/orders/flows/update/open/OpenCompositeOrderPieceService.java

```diff
@@ -72,89 +71,79 @@ public OpenCompositeOrderPieceService(PurchaseOrderStorageService purchaseOrderS
       .compose(v -> pieceStorageService.insertPiece(piece, requestContext));
   }
 
-  public Future<Void> updatePieceRecord(Piece piece, RequestContext requestContext) {
-    return purchaseOrderStorageService.getCompositeOrderByPoLineId(piece.getPoLineId(), requestContext)
-      .compose(order -> titlesService.getTitleById(piece.getTitleId(), requestContext)
-        .compose(title -> protectionService.isOperationRestricted(title.getAcqUnitIds(), ProtectedOperationType.UPDATE, requestContext)))
+  public Future<Piece> updatePiece(Piece piece, RequestContext requestContext) {
+    logger.debug("updatePiece:: Updating piece");
+    return titlesService.getTitleById(piece.getTitleId(), requestContext)
+      .compose(title -> protectionService.isOperationRestricted(title.getAcqUnitIds(), ProtectedOperationType.UPDATE, requestContext))
       .compose(v -> inventoryItemManager.updateItemWithPieceFields(piece, requestContext))
       .compose(vVoid -> pieceStorageService.getPieceById(piece.getId(), requestContext))
       .compose(pieceStorage -> {
-        Piece.ReceivingStatus receivingStatusUpdate = piece.getReceivingStatus();
-        Piece.ReceivingStatus receivingStatusStorage = pieceStorage.getReceivingStatus();
-        boolean isReceivingStatusChanged = receivingStatusStorage.compareTo(receivingStatusUpdate) != 0;
-
+        var receivingStatusUpdate = piece.getReceivingStatus();
+        var receivingStatusStorage = pieceStorage.getReceivingStatus();
+        boolean isReceivingStatusChanged = !receivingStatusStorage.equals(receivingStatusUpdate);
         if (isReceivingStatusChanged) {
           piece.setStatusUpdatedDate(new Date());
         }
-
         return pieceStorageService.updatePiece(piece, requestContext)
-          .map(ok -> {
-            JsonObject messageToEventBus = new JsonObject();
-            messageToEventBus.put("poLineIdUpdate", piece.getPoLineId());
-            logger.debug("receivingStatusStorage -- {}", receivingStatusStorage);
-            logger.debug("receivingStatusUpdate -- {}", receivingStatusUpdate);
-
+          .compose(v -> {
+            logger.debug("updatePiece:: receivingStatusStorage - {}", receivingStatusStorage);
+            logger.debug("updatePiece:: receivingStatusUpdate - {}", receivingStatusUpdate);
             if (isReceivingStatusChanged) {
+              var messageToEventBus = JsonObject.of("poLineIdUpdate", piece.getPoLineId());
               receiptStatusPublisher.sendEvent(MessageAddress.RECEIPT_STATUS, messageToEventBus, requestContext);
             }
-            return null;
+            return Future.succeededFuture();
           })
-          .onFailure(e -> logger.error("Error updating piece by id to storage {}", piece.getId(), e));
+          .onFailure(e -> logger.error("updatePiece:: Error updating piece by id to storage {}", piece.getId(), e));
```

Comment:
Probably it would be nicer to keep the same pattern for all logs with the corresponding method names. Apart from this no other solid reason to keep name this unless you think removing it would be a better option

Reply:
For me it is better to removing as well because from stack trace we can easily find method and line in source code.
Also in all other places we don't have method name in exception logs, so approach will not be consistent with existing implementation

Reply:
Got it, should be good now